homeassistant/components/geo_json_events/manager.py

```diff
@@ -23,45 +32,66 @@ class GeoJsonFeedEntityManager:
     def __init__(
         self,
         hass: HomeAssistant,
-        scan_interval: timedelta,
-        coordinates: tuple[float, float],
-        url: str,
-        radius_in_km: float,
+        config_entry: ConfigEntry,
     ) -> None:
         """Initialize the GeoJSON Feed Manager."""
-
-        self._hass = hass
+        self._hass: HomeAssistant = hass
+        self._config_entry: ConfigEntry = config_entry
         websession = aiohttp_client.async_get_clientsession(hass)
-        self._feed_manager = GenericFeedManager(
+        self._feed_manager: GenericFeedManager = GenericFeedManager(
             websession,
             self._generate_entity,
             self._update_entity,
             self._remove_entity,
-            coordinates,
-            url,
-            filter_radius=radius_in_km,
+            (
+                config_entry.data[CONF_LATITUDE],
+                config_entry.data[CONF_LONGITUDE],
+            ),
+            config_entry.data[CONF_URL],
+            filter_radius=config_entry.data[CONF_RADIUS],
+        )
+        self._config_entry_id: str = config_entry.entry_id
+        self._scan_interval: timedelta = timedelta(
+            seconds=config_entry.data[CONF_SCAN_INTERVAL]
         )
-        self._scan_interval = scan_interval
-        self.signal_new_entity = (
-            f"{DOMAIN}_new_geolocation_{coordinates}-{url}-{radius_in_km}"
+        self._track_time_remove_callback: Callable[[], None] | None = None
+        self.listeners: list[Callable[[], None]] = []
+        self.signal_new_entity: str = (
+            f"{DOMAIN}_new_geolocation_{self._config_entry_id}"
         )
 
     async def async_init(self) -> None:
         """Schedule initial and regular updates based on configured time interval."""
 
+        await self._hass.config_entries.async_forward_entry_setups(
+            self._config_entry, PLATFORMS
+        )
```

Comment:
I think this should be inside `__init__.async_setup_entry`, before calling `async_init`, to be more in line with other integrations.

Reply:
OK, I moved the code.